### PR TITLE
Plone 5.1.x support

### PR DESCRIPTION
**Pull request by maethu**
- Implement support for plone.app.multilingual 5.x
- Implement constrain types the plone 5 way
- Use ftw.testbrowser for browser tests
- Fix several assertions for plone 5

### CODE REVIEW

**Comment by maethu** [comment]
@jone Sorry it's a lot, but it looks like it works just fine again. 

Well there are some style issue, but we could fixt that later.

**Comment by busykoala** [comment]
It is working installed in webdemo_5... Thanks

File: ftw/inflator/browser/overview_plone5.pt [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+                         tal:content="customization/product" />
+            is up and running.
+        </h1>
+        <ul>
+            <tal:block tal:condition="sites">
+                <tal:one tal:condition="python:len(sites) == 1">
+                    <tal tal:repeat="site sites">
+                        <span class="prominent">
+                            <!--<img tal:attributes="src string:${site/absolute_url}/logoIcon.png"
+                                 height="16" width="16"/>-->
```
Comment (jone) [inline ftw/inflator/browser/overview_plone5.pt]: ✂️ ?

File: ftw/inflator/overrides.zcml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    xmlns="http://namespaces.zope.org/zope"
+    xmlns:browser="http://namespaces.zope.org/browser"
+    xmlns:i18n="http://namespaces.zope.org/i18n"
+    xmlns:zcml="http://namespaces.zope.org/zcml"
+    i18n_domain="ftw.inflator">
+
+    <browser:page
+        zcml:condition="not-have plone-5"
+        for="OFS.interfaces.IApplication"
+        name="plone-overview"
```
Comment (jone) [inline ftw/inflator/overrides.zcml]: Duplicate: https://github.com/4teamwork/ftw.inflator/pull/36/files#diff-5c3e52bda19d593850b74bcdb0ef46ceL25

I think this should be possible without an `overrides.zcml` at all. It did work on Plone 4; why shouldn't it work on Plone 5?

File: test-plone-4.3.x.cfg [hunk 3]
```diff
@@ -1,7 +1,13 @@
...
     sources.cfg
 
 package-name = ftw.inflator
+
+eggs +=
+    ftw.inflator[plone4_multilingual]
+
+[versions]
+plone.app.multilingual = <2a
+archetypes.multilingual = <2a
```
Comment (jone) [inline test-plone-4.3.x.cfg]: Should we move that to https://github.com/4teamwork/ftw-buildouts/blob/master/test-versions-plone-4.cfg ?

File: test-plone-5.1.x.cfg [hunk 4]
```diff
@@ -0,0 +1,9 @@
+[buildout]
+extends =
+    https://raw.github.com/4teamwork/ftw-buildouts/master/test-plone-5.1.x.cfg
+    sources.cfg
+
+package-name = ftw.inflator
+
+[versions]
+plone.app.multilingual = 5.3.2
```
Comment (jone) [inline test-plone-5.1.x.cfg]: Should we move that to https://github.com/4teamwork/ftw-buildouts/blob/master/test-versions-plone-5.cfg ?

File: ftw/inflator/overrides.zcml (hunk 2 above)
Comment (maethu) [reply ftw/inflator/overrides.zcml]: Unfortunately browserlayers are not yet ready. 

The request only provides the following:
``` 
(Pdb) tuple(self.request.__provides__)
(<InterfaceClass zope.publisher.interfaces.browser.IDefaultBrowserLayer>, <InterfaceClass zope.publisher.interfaces.browser.IBrowserRequest>, <InterfaceClass zope.annotation.interfaces.IAttributeAnnotatable>)
```

and adding any sort of layer to this `browser:page` has no effect.

I tried `zope.publisher.interfaces.browser.IDefaultBrowserLayer` and a interface which inherits from `zope.publisher.interfaces.browser.IDefaultBrowserLayer`. Both had no effect.

File: test-plone-4.3.x.cfg (hunk 3 above)
Comment (maethu) [reply test-plone-4.3.x.cfg]: moved to https://github.com/4teamwork/ftw-buildouts/pull/148

File: test-plone-5.1.x.cfg (hunk 4 above)
Comment (maethu) [reply test-plone-5.1.x.cfg]: Moved to https://github.com/4teamwork/ftw-buildouts/pull/148 but I let this specific version pinning in here

**Comment by maethu** [comment]
@jone Tests are failing until https://github.com/4teamwork/ftw-buildouts/pull/148/files is merged.

File: ftw/inflator/overrides.zcml (hunk 2 above)
Comment (jone) [reply ftw/inflator/overrides.zcml]: 👍

File: test-plone-5.1.x.cfg (hunk 4 above)
Comment (jone) [reply test-plone-5.1.x.cfg]: 👍 merged

File: test-plone-4.3.x.cfg (hunk 3 above)
Comment (jone) [reply test-plone-4.3.x.cfg]: 👍 merged

**Pull request merged**